### PR TITLE
chore: mark api classes with `@final`

### PR DESCRIPTION
**Pull request by kumaraditya303**
Closes https://github.com/microsoft/playwright-python/issues/852

### CODE REVIEW

File: scripts/generate_async_api.py [hunk 1]
```diff
@@ -40,15 +40,16 @@ def generate(t: Any) -> None:
...
     else:
-        base_sync_class = base_class
-    print(f"class {class_name}({base_sync_class}):")
+        base_async_class = base_class
+    print("@final")
+    print(
+        f"class {class_name}({base_async_class}): {'# type: ignore' if base_async_class== 'JSHandle' else ''}"
+    )
     print("")
-    print(f"    def __init__(self, obj: {class_name}Impl):")
```
Comment (kumaraditya303) [inline scripts/generate_async_api.py:50]: This was redundant so removed

File: scripts/generate_async_api.py [hunk 2]
```diff
@@ -40,15 +40,16 @@ def generate(t: Any) -> None:
...
     elif base_class in ["ChannelOwner", "object"]:
-        base_sync_class = "AsyncBase"
+        base_async_class = "AsyncBase"
     else:
-        base_sync_class = base_class
-    print(f"class {class_name}({base_sync_class}):")
+        base_async_class = base_class
+    print("@final")
+    print(
+        f"class {class_name}({base_async_class}): {'# type: ignore' if base_async_class== 'JSHandle' else ''}"
```
Comment (kumaraditya303) [inline scripts/generate_async_api.py:50]: `# type: ignore` is for `ElementHandle` as it inherits from `JSHandle`

Comment (mxschmitt) [reply scripts/generate_async_api.py:50]: is the `@final` thing only for linting or does it actually throw when you try to extend from it again?

Comment (symonk) [reply scripts/generate_async_api.py:50]: don't think its `runtime_checkable`, if it's only for mypy this is fine, _enforce_ it through other means (__init_subclass__? or metaclass I guess) but not sure if that's what you want to enforce

**Comment by kumaraditya303** [comment]
> Since `@final` seems not common and does not provide a huge benefit we decided to not land this to reduce the complexity overhead.

Huge benefit here = better types, also `@final` was added on high user demand to typing hence it has a dedicated PEP

**Comment by mxschmitt** [comment]
Closing to keep the complexity of the generator logic low and there was no user reports before. But still thank you for your contribution!

**Pull request closed without merging**